go/providers.rst

```diff
@@ -1,34 +1,172 @@
+| The cover variables used in these sources.                                                       |
++--------------------------------+-----------------------------------------------------------------+
+| :param:`cgo_deps`              | :type:`depset(cc_library)`                                      |
++--------------------------------+-----------------------------------------------------------------+
+| The direct cgo dependencies of this library.                                                     |
++--------------------------------+-----------------------------------------------------------------+
+| :param:`gc_goopts`             | :type:`tuple(string)`                                           |
++--------------------------------+-----------------------------------------------------------------+
+| Go compilation options that should be used when compiling these sources.                         |
+| In general these will be used for *all* sources of any library this provider is embedded into.   |
++--------------------------------+-----------------------------------------------------------------+
+
 
 GoBinary
 ~~~~~~~~
 
-**TODO**: More information
+This is the provider returned by anything that links an executable go binary. This includes both
+the go_binary_ rule and the go_test_ rule.
+
++--------------------------------+-----------------------------------------------------------------+
+| **Name**                       | **Type**                                                        |
++--------------------------------+-----------------------------------------------------------------+
+| :param:`executable`            | :type:`File`                                                    |
++--------------------------------+-----------------------------------------------------------------+
+| The default executable binary produced by this rule.                                             |
+| This is the file that would be invoked using ``bazel run``.                                      |
++--------------------------------+-----------------------------------------------------------------+
+| :param:`static`                | :type:`File`                                                    |
++--------------------------------+-----------------------------------------------------------------+
+| The binary compiled in `static linking`_ mode.                                                      |
++--------------------------------+-----------------------------------------------------------------+
+| :param:`race`                  | :type:`File`                                                    |
++--------------------------------+-----------------------------------------------------------------+
+| The binary compiled with the `race detector`_ turned on.                                            |
++--------------------------------+-----------------------------------------------------------------+
```

Comment:
Consider adding a short example. Maybe a rule that depends on a `go_library` and writes import paths of transitive dependencies to a file.

Reply:
I have been trying to avoid having larger examples in the docs, I think what we ought to do is clean up our tests vs examples, such that the examples really show how to use the rules, and then when that is done we can add links to the examples from the docs.